tests/utils/test_shell_util.py

```diff
@@ -490,6 +480,15 @@ def _get_command_line(pid):
             pass
         return "[PID: {0}] UNKNOWN".format(pid)
 
+    @staticmethod
+    def _wait_for(predicate):
+        start_time = datetime.datetime.now()
+        while RunCommandTestCase._to_seconds(datetime.datetime.now() - start_time) < 30:
```

Comment:
So this one test could take up to 30 seconds? Any reason for waiting that long?
Granted, this should only happen if the threads were unable to start for some reason, which is not a normal case.

Reply:
No, the test takes 0.05-0,0.08 s (my local machine). 30 s is the timeout if things go wrong

Reply:
Yes, I was wondering if that timeout could be shorter so we fail faster, but I realized this probably won't happen often.

Reply:
Oh, I think I misunderstood your point. 30 s is probably too generous. I am trying to avoid the test being noisy if, for example, one is running the unit test in a laptop while using a chat app that is taking most of the cpu.

Reply:
Sounds like a bad chat app :smile:

Reply:
And probably we should add a verification to the entire suite to let us know if the run takes more than some given time. A bad mock, a regression in prod/test code may go unnoticed until somebody stumbles into a slow test run and realizes something is not correct.

Reply:
I can look into the Github Actions docs; I feel like this sort of thing should be easy to automate as a separate, not required job.